getting_started/scripting/gdscript/gdscript_types_intro.rst

```diff
@@ -0,0 +1,413 @@
+``PlayerController`` attached to it. You use the ``on_body_entered``
+signal to detect the collision. With typed code, the body you detect is
+going to be a generic ``PhysicsBody2D``, and not your
+``PlayerController`` on the ``_on_body_entered`` callback.
+
+You can check if this ``PhysicsBody2D`` is your Player with the ``as``
+casting keyword, and using the colon ``:`` again to force the variable
+to use this type. This forces the variable to stick to the
+``PlayerController`` type:
+
+::
+
+    func _on_body_entered(body : PhysicsBody2D) -> void:
+    var player := body as PlayerController
+    if not player:
+        return
+    player.damage()
+
+As we’re dealing with a custom type, if the ``body`` doesn’t extend
+``PlayerController``, the ``player``\ variable will be set to ``null``.
+We can use this to check if the body is the player or not. We will also
+get full autocompletion on the player variable thanks to that cast.
+
+.. note::
+
+    If you try to cast with a built-in type and it fails, Godot will throw an error.
+
+Safe lines
+^^^^^^^^^^
+
+You can also use casting is to ensure safe lines. Safe lines are a new
+tool in Godot 3.1 to tell you when ambiguous lines of code are
+type-safe. As you can mix and match typed and dynamic code, at times,
+Godot doesn’t have enough information to if an instruction will trigger
+an error or not at runtime.
+
+This happens when you get a child node. Let’s take a timer for example:
+with dynamic code, you can get the node with ``$Timer``. GDscript
+supports `duck-typing <https://stackoverflow.com/a/4205163/8125343>`__,
+so even if your timer is of type ``Timer``, it is also a ``Node`` and an
+``Object``, two classes it extends. With dynamic GDscript, you also also
+don’t care about the node’s type as long as it has the methods you need
+to call.
+
+You can use casting to tell Godot the type you expect when you get a
+node: ``($Timer as Timer)``, ``($Player as KinematicBody2D)``, etc.
+Godot will ensure the type works and if so, the line number will turn
+green at the left of the script editor.
+
+.. figure:: ./img/typed_gdscript_safe_unsafe_line.png
+   :alt: Safe vs Unsafe Line
+
+   Safe vs Unsafe Line
+
+.. note::
+
+    You can turn off safe lines or change their color in the editor settings.
+
+Define the Return Type of a Function with the Arrow ->
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To define the return type of a function, write a dash and a right angle
+bracket ``->`` after it’s declaration, followed by the return type:
+
+::
+
+    func _process(delta : float) -> void:
+        pass
+
+The type ``void`` means the function does not return anything. You can
+use any type as with variables:
+
+::
+
+    func hit(damage : float) -> bool:
+        health_points -= damage
+        return health_points <= 0
+
+You can also use your own nodes as return types:
+
+::
+
+    """Inventory.gd"""
+
+    """Adds an item to the inventory and returns it"""
```

Comment:
I'm not sure anyone else agrees, but these one-line comments using multi-line string syntax bothers me. Why not just start the lines with `# ` instead?

Reply:
Highlighting: string highlighting is a lot more visible, while comments display in a washed out grey tone.